File: elixir/plug-cowboy/lib/plug_cowboy_app/controllers/prime_controller.ex
defmodule Controllers.PrimeController do
  import Services.Prime
  import Utils.DelayedTask
  alias Models.LoopStatus
  @moduledoc false

  def primes(percentage, delay, top) do
    times = start(fn -> process(top) end, percentage, delay)
    LoopStatus.ok(times > 0, times)
  end

end

File: stacks/nodejs/nestjs/src/controllers/prime/prime.controller.ts
import { Controller, Get, Param, Query } from '@nestjs/common';
import { LoopStatus } from '../../models/loop-status';
import { PrimeService } from '../../services/prime/prime.service';
import { loopCall } from '../../utils';

@Controller()
export class PrimeController {
  constructor(private readonly primeService: PrimeService) {}

  @Get('/prime/:top')
  async getStatus(
    @Query('delay') delayInput,
    @Query('percentage') percentageInput,
    @Param('top') top
  ): Promise<LoopStatus> {
    return loopCall(delayInput, percentageInput, () => this.primeService.process(top));
  }
}